desmod/workspacesync/s3.py

```diff
@@ -0,0 +1,59 @@
+from concurrent.futures import ThreadPoolExecutor
+
+import boto3
+import getpass
+import os
+
+
+class S3Sync(object):
```

Comment:
The main reason I went with this approach to perhaps have common set of methods for multiple backend synchronizers, which could register themselves and perhaps one day be called in a loop like so:

```python
for cls in sync_registry:
    cls(config, artifacts).sync()
```

That said, that's a lot of hand waving, and we only have one sync backend now, so I probably shouldn't be solving problems that don't exist yet.